### PR TITLE
[GitHub] Add ability to view pull request comments

### LINKED ISSUE #12170: [GitHub] Ability to see all Comments / full Conversation of a pull request
### Extension

https://www.raycast.com/raycast/github

### Description

when opening the pull request I can't see comments on the pull request, I can add one but not see all of them, which doesn't make much sense.

### Who will benefit from this feature?

-> short from everyone

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by ramsaysewell**
## Description

Closes #12170.

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by ramsaysewell** [comment]
Fixes #11626

**Comment by LitoMore** [comment]
Could you please revert those unrelated changes from your formatter?

I can see most of markdown files were foramtted into another style.

**Comment by ramsaysewell** [comment]
> Could you please revert those unrelated changes from your formatter?
> 
> I can see most of markdown files were foramtted into another style.

I don't have any IDE specific formatting rules, the .prettierrc handles formatting style. I'm unsure how this _did_ affect the already existing lines however.

**Comment by LitoMore** [comment]
> I don't have any IDE specific formatting rules, the .prettierrc handles formatting style. I'm unsure how this did affect the already existing lines however.

No. Please double-check local configurations. You can try to run `npm run fix-lint` to verify the auto-fix behavior. Those lines should not be formatted.

**Comment by ramsaysewell** [comment]
> > I don't have any IDE specific formatting rules, the .prettierrc handles formatting style. I'm unsure how this did affect the already existing lines however.
> 
> No. Please double-check your local configurations. You can try `npm run fix-lint` to verify the auto-fix behavior. Those lines should not be formatted.

@LitoMore The only formatting difference shown in my changes is the `printWidth` key in .prettierrc which has been set at 120. The linter must've formatted these existing lines to the set printWidth. I'm unaware if this printWidth value has only been recently set.

**Comment by LitoMore** [comment]
@ramsaysewell It should be related to your IDE default settings. I've tried your code with `npm run fix-lint`, which does not automatically format line breaks.

As I said, you can use those lines back and run `npm run fix-lint` to verify the behavior.

**Comment by thomaslombart** [comment]
Hey there! I'll be on vacation next week, so I'll review this PR the following week. @pernielsentikaer, feel free to handle this one if you have time.

File: extensions/github/CHANGELOG.md
Comment (thomaslombart) [inline extensions/github/CHANGELOG.md:1]: Could you clean this file up as @LitoMore requested? 🙂

File: extensions/github/CHANGELOG.md [hunk 1]
```diff
@@ -1,16 +1,26 @@
 # GitHub Changelog
 
+## [Add "Show Pull Request Comments"] - 2024-06-18
+
+- adds ability to view pull request comments
+
```
Comment (thomaslombart) [inline extensions/github/CHANGELOG.md:6]: Could you group both features in one PR (comments + filterable sections)

File: extensions/github/src/api/pullRequest.graphql
Comment (thomaslombart) [inline extensions/github/src/api/pullRequest.graphql:1]: These additions should probably be done as part of another fragment (e.g. like on `PullRequestDetailsFields`), as requesting more data will increase the likelihood of having timeout errors for people with a lot of pull requests.

File: extensions/github/src/components/PullRequestComments.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  mutateList?: MutatePromise<PullRequestFieldsFragment[] | undefined> | ReturnType<typeof useMyPullRequests>["mutate"];
+};
+
+export const PullRequestComments = ({ pullRequest, viewer, mutateList }: PullRequestCommentsProps) => {
+  return (
+    <List isShowingDetail>
+      {pullRequest.comments.nodes ? (
+        pullRequest.comments.nodes
+          .filter((c) => c !== null)
+          .sort((a, b) => new Date(b!.createdAt).valueOf() - new Date(a!.createdAt).valueOf())
```
Comment (thomaslombart) [inline extensions/github/src/components/PullRequestComments.tsx:24]: Isn't it possible to sort that directly in GitHub's graphql API?

File: extensions/github/src/components/PullRequestListItem.tsx [hunk 3]
```diff
@@ -82,6 +83,13 @@ export default function PullRequestListItem({ pullRequest, viewer, mutateList }:
             icon={Icon.Sidebar}
             target={<PullRequestDetail initialPullRequest={pullRequest} viewer={viewer} mutateList={mutateList} />}
           />
+          {numberOfComments > 0 && (
+            <Action.Push
+              title="Show Comments"
+              icon={Icon.Message}
+              target={<PullRequestComments viewer={viewer} mutateList={mutateList} pullRequest={pullRequest} />}
+            />
+          )}
```
Comment (thomaslombart) [inline extensions/github/src/components/PullRequestListItem.tsx:92]: This should be added below `Open in Browser` as it'll break people using ⌘ + ⏎ to open the PR in the browser.

File: extensions/github/src/helpers/menu-bar.ts
Comment (thomaslombart) [inline extensions/github/src/helpers/menu-bar.ts:1]: Even though the two menu bar commands share similarities for preferences, I would still keep the code duplicated to make it easier to maintain and understand.

File: extensions/github/src/helpers/pull-request.ts [hunk 4]
```diff
@@ -131,3 +132,48 @@ export function getReviewDecision(reviewDecision?: PullRequestReviewDecision | n
...
+    case "ROCKET":
+      return "🚀";
+    case "EYES":
+      return "👀";
+    default:
+      return "";
+  }
+}
+// __typename?: "Reaction" | undefined;   content: ReactionContent;   id: string;
+export function getReducedReactions(
```
Comment (thomaslombart) [inline extensions/github/src/helpers/pull-request.ts:159]: Let's call that `getReactionsCount` to make it more explicit.

**Comment by pernielsentikaer** [comment]
Are you still around @ramsaysewell? Marking as Draft until you're ready 🙂

**Comment by jfkisafk** [comment]
Hello @pernielsentikaer / @thomaslombart / @ramsaysewell , I have resolved https://github.com/raycast/extensions/issues/11626 as part of PR: https://github.com/raycast/extensions/pull/13908.

**Pull request closed without merging**